app/src/angular/services/web3.service.js

```diff
@@ -305,16 +253,13 @@ function Web3Service($rootScope, $window, $q, $timeout, $log, $http, $httpParamS
         }
 
 
-        static waitForTicket(defer, method, args, contract, baseFn) {
-            Web3Service.q.push({ method: method, args: args, contract: contract, baseFn: baseFn }, (promise) => {
-                $log.info("handle response", method);
-                promise.then((response) => {
-                    $log.info("method response", method, response);
-                    defer.resolve(response)
-                }).catch((error) => {
-                    $log.error("method response error", method, error);
-                    defer.reject(error);
-                });
+        static waitForTicket(defer, method, args, contractAddress, contractMethod) {
+            RPCService.makeCall('waitForWeb3Ticket', { method, args, contractAddress, contractMethod }).then((response) => {
+                $log.info("method response", method, response);
+                defer.resolve(response)
+            }).catch((error) => {
+                $log.error("method response error", method, error);
+                defer.reject(error);
             });
         }
     };
```

Comment:
question: why mix of Q and Bluebird for promises throughout the app?  can we refactor to native promises or async?